### PR TITLE
x11-libs/fox: Fix ISO C++17 does not allow register storage class spe…

### PR DESCRIPTION
**Pull request by listout**
…cifier

Closes: https://bugs.gentoo.org/898866

### CODE REVIEW

**Review by juippis — APPROVED** [review]
Can confirm the build succeeds with clang:16 and/or musl+clang:16.

@zx2c4 any comments? Could you merge if you ack this PR?

**Comment by juippis** [comment]
@listout oh I forgot to note though:
Per [GLEP-66](https://www.gentoo.org/glep/glep-0066.html#commit-messages) the commit summary must be <70 chars. You've probably seen me update them to be shorter while merging, but figured you want to know why and prevent that from happening in future!

**Comment by listout** [comment]
Thanks for letting me know. I'll keep this is mind while committing in future.

**Comment by listout** [comment]
@juippis ping :)

File: x11-libs/fox/fox-1.7.67-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	"${FILESDIR}"/"${PN}"-1.7.67-no-truetype.patch
+	"${FILESDIR}"/"${PN}"-1.7.67-pthread_rwlock_prefer_writer_np-musl.patch
+)
+
+src_prepare() {
+	default
+
+	sed -i '/#define REXDEBUG 1/d' lib/FXRex.cpp || die "Unable to remove spurious debug line."
+	sed -i -e "s:windows::" Makefile.am || die
+	sed -i -e 's/register //g' lib/*.cpp || die "Unable remove register keywords from sources under lib folder"
```
Comment (thesamesam) [inline x11-libs/fox/fox-1.7.67-r2.ebuild:46]: I'd really prefer a patch for this but Fox is dead, I think.

**Pull request closed without merging**